eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -243,6 +245,19 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, wallet: OnChainA
         d.channels.values.toSet[ActorRef].foreach(_ ! INPUT_DISCONNECTED) // we deduplicate with toSet because there might be two entries per channel (tmp id and final id)
         gotoConnected(connectionReady, d.channels)
 
+      case Event(msg: OnionMessage, _: ConnectedData) =>
+        if (nodeParams.features.hasFeature(Features.OnionMessages)) {
+          OnionMessages.process(nodeParams.privateKey, msg) match {
+            case OnionMessages.DropMessage(reason) =>
+              log.debug(s"dropping message from ${remoteNodeId.value.toHex}: ${reason.toString}")
+            case OnionMessages.RelayMessage(nextNodeId, dataToRelay) =>
+              context.spawnAnonymous(MessageRelay(switchboard, nextNodeId, dataToRelay))
```

Comment:
From an architecture's point of view, I'm not sure the `Peer` should be spawning these actors. The issue with that is that if the `Peer` is stopped, all its children are stopped, which could terminate a message relay in the middle of its operations without any warning. This isn't an issue right now, but when we automatically stop idle peers with which we have no channels, there is a risk that this will bite us.

I think the `MessageRelay` actors should be spawned by the `Switchboard`, which is a singleton that should always stay alive. I think it's the `Switchboard` that should handle a `SendOnionMessage` request and create the child `MessageRelay` actor. Similarly, it means that `sendOnionMessage` in `Eclair.scala` shouldn't spawn the `MessageRelay` actor but instead just send a message to the `Switchboard`.

Reply:
Done